sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -2098,19 +2099,22 @@ private void SetNameFieldsIfNull()
                     // This also makes sure that we fail fast if file doesn't exist.
                     var blobProperties = await GetPropertiesInternal(conditions: conditions, async, cancellationToken).ConfigureAwait(false);
 
-                    var expectedETag = blobProperties.Value.ETag;
-                    var readConditions = allowModifications ? new BlobRequestConditions() : new BlobRequestConditions { IfMatch = expectedETag };
+                    var etag = blobProperties.Value.ETag;
+                    var readConditions = conditions;
+                    if (!allowModifications)
+                    {
+                        readConditions = readConditions?.WithIfMatch(etag) ?? new BlobRequestConditions { IfMatch = etag };
```

Comment:
So now the logic is "force you to track the ETag unless you specifically opted to live dangerously"?  I think that makes sense.